packages/forms/src/UIForm/fields/ListView/ListView.component.test.js

```diff
@@ -0,0 +1,285 @@
+			{ name: 'Andorra', value: 'country4' },
+			{ name: 'Angola', value: 'country5' },
+			{ name: 'Anguilla', value: 'country6' },
+		],
+	};
+	const noItemsSchema = {
+		description: 'This is the ListView field',
+		disabled: false,
+		emptyLabel: 'No element',
+		noResultLabel: 'No element matching your filter',
+		placeholder: 'Search filter',
+		required: true,
+		title: 'Countries',
+		titleMap: [],
+	};
+
+	beforeEach(() => {
+		props = {
+			id: 'my-list-view',
+			isValid: true,
+			errorMessage: 'This is wrong',
+			onChange: jest.fn(),
+			onFinish: jest.fn(),
+			schema,
+			value: ['country2', 'country3'],
+		};
+	});
+
+	describe('render', () => {
+		it('should render ListView', () => {
+			// when
+			const wrapper = shallow(<ListView {...props} />);
+
+			// then
+			expect(wrapper.node).toMatchSnapshot();
+		});
+
+		it('should render no items message', () => {
+			// when
+			const wrapper = mount(<ListView {...props} schema={noItemsSchema} />);
+
+			// then
+			expect(wrapper.find('.tc-listview').at(0).node).toMatchSnapshot();
+		});
+	});
+
+	describe('componentWillReceiveProps', () => {
+		it('should update items on props.value change', () => {
+			// given
+			const node = document.createElement('div');
+			// eslint-disable-next-line react/no-render-return-value
+			const instance = ReactDOM.render(<ListView {...props} />, node);
+			expect(instance.state).toMatchSnapshot();
+
+			const allValues = props.schema.titleMap.map(option => option.value);
+
+			// when : trigger a props update
+			ReactDOM.render(<ListView {...props} value={allValues} />, node);
+
+			// then
+			expect(instance.state).toMatchSnapshot();
+		});
+
+		it('should update items on props.schema change', () => {
+			// given
+			const node = document.createElement('div');
+			// eslint-disable-next-line react/no-render-return-value
+			const instance = ReactDOM.render(<ListView {...props} />, node);
+			expect(instance.state).toMatchSnapshot();
+
+			const allValues = alternativeSchema.titleMap.map(option => option.value);
+
+			// when : trigger a props update
+			ReactDOM.render(<ListView
+				{...props}
+				schema={alternativeSchema}
+				value={allValues}
+			/>, node);
+
+			// then
+			expect(instance.state).toMatchSnapshot();
+		});
+	});
+
+	describe('filter', () => {
+		it('should switch to input mode on search button click', () => {
+			// given
+			const wrapper = mount(<ListView {...props} />);
+			const initialHeader = wrapper.find('.tc-listview-header').at(0);
+			expect(initialHeader.node).toMatchSnapshot();
+
+			// when
+			initialHeader.find('button').at(0).simulate('click');
+
+			// then
+			expect(wrapper.find('.tc-listview-header').at(0).node).toMatchSnapshot();
+		});
+
+		it('should filter items on input change', () => {
+			// given
+			const wrapper = mount(<ListView {...props} />);
+
+			// when
+			filter(wrapper, 'ia');
+
+			// then
+			expect(wrapper.find('.tc-listview-items').at(0).node).toMatchSnapshot();
```

Comment:
snapshots doesn't test anything here.

Reply:
Snapshots test that the suggestions in DOM are filtered. If I break the filter, the snapshot will add 2 more values in items list